OCS/deploy-ocs.sh

```diff
@@ -4,15 +4,15 @@ LOCALNAMESPACE="local-storage"
 
 # export is required for envsubst
 export rook_override="${rook_override:-false}"
-export public_network="${public_network:-1.1.1.1/24}"
-export cluster_network="${cluster_network:-1.1.1.1/24}"
+export public_network="$(ip r | awk '/dev\ baremetal\ proto\ kernel\ scope\ link\ src/ {print $1}')"
+export cluster_network="$(ip r | awk '/dev\ provisioning\ proto\ kernel\ scope\ link\ src/ {print $1}')"
 
 # Provide disks to use for mon and osd pvcs
 export cluster="${cluster:-mycluster}"
 # Size number for mon pvcs
 export mon_size="${mon_size:-5}"
 # List of /dev/* disks to use for osd, separated by comma
-export osd_devices="${osd_devices:-}"
+export osd_devices="$(lsblk -p -d -n -o name -I8,259 | tail -n +2 | paste -s -d ',')"
 # Size number for osd pvcs
```

Comment:
I know this works on real BM , will it work in a virtual env?

Reply:
Yep, works on both:

bare metal:
```shell
[cloud-user@r640-u01 ~]$ lsblk -p -d -n -o name -I8,259 | tail -n +2 | paste -s -d ','
/dev/nvme2n1,/dev/nvme1n1,/dev/nvme0n1,/dev/nvme3n1
```
virt:
```shell
[cloud-user@rhhi-node-worker-0 ~]$  lsblk -p -d -n -o name -I8,259 | tail -n +2 | paste -s -d ','
/dev/sdb,/dev/sdc,/dev/sdd,/dev/sde
```

Reply:
So the assumption is that the first disk willbe the OS and the remaining are for OCS.  Maybe add a comment saying that?

Reply:
i suggest to default to those values only when  the env variables are blank, so that we still let the opportunity for the end user to provide them, if wanted

Reply:
> i suggest to default to those values only when the env variables are blank, so that we still let the opportunity for the end user to provide them, if wanted

Done, updated the vars to be autogenerated only if they aren't set.